proposals/array_set.json

```diff
@@ -0,0 +1,215 @@
+{
+    "id": "array_set",
+    "summary": "Change the content of an array (insert, remove, update)",
+    "description": "Allows to insert into, remove from or update an array.\n\nAll labels get discarded and the array indices are always a sequence of numbers with the step size of 1 and starting at 0.",
```

Comment:
Personally, I'm always confused with ``splice`` in JS and PHP. I'm programming both and always need to check the docs, which function to use and whether splice is what I need to use. replace would not cover the full range of functionality, IMHO. Some synonyms I think could work: modify, transform, change, alter (ordered by my preference). @soxofaan What do you prefer?

Reply:
Indeed, I'm also confused by "splice" in JS and PHP, especially because you also have "slice" which sounds almost the same, and acts almost the same.

While looking up how Scala names this functionality, I found another possibility I prefer now: `array_patch`. If you are familiar with version control the verb "patch" should be synonym for an operation where you add, replace or remove a subset and leave the rest as-is. Perfect fit if you ask me :smile:

Reply:
Thought exactly the same regarding slice. 😂

I must admit that I think I'm confident with some version control systems but would not have understood array_patch without your explanation. I could have made the connection to HTTP Patch though, but who does that from our "customers"? (Also I just renamed it to modify before reading your post. Does that work for you, too?)

Reply:
`array_modify` is more generic and less self-descriptive in my opinion. 
`array_patch` more clearly communicates to me that it is about modifying only a subset of items, but it requires indeed that you are familiar with the verb "patch"

It would be nice if we could collect some more viewpoints on this. I'll try to collect some feedback from our internal users.

Reply:
Like below, I'll also request reviews from @aljacob, @lforesta etc.

Reply:
I got some feedback from internal poll about the name and 3/3 responses preferred `array_modify`

Reply:
Great, I'll change the name to array_modify then.